### PR TITLE
mixed-content: Check that 'http://127.0.0.1/' is not blocked as mixed content

### PR DESCRIPTION
**Pull request by poiru**
According to the spec, content from loopback addresses should no longer
be treated as mixed content even in secure origins. See:
- w3c/webappsec-mixed-content@349501c
- https://w3c.github.io/webappsec-secure-contexts/#is-origin-trustworthy

The added tests pass in Chrome 53 and later, see:
- https://chromium.googlesource.com/chromium/src.git/+/130ee686fa00b617bfc001ceb3bb49782da2cb4e

They will also soon pass in Firefox, see:
- https://bugzilla.mozilla.org/show_bug.cgi?id=903966

It is unclear if HTTPS origins should be able to use workers and WebSocket
connections through a loopback HTTP address. They are not supported in Chrome
(whether this is intentional or not is uncertain) so I chose not to include
`websocket-request` and `worker-request` checks for the `loopback-allowed`
case.

Also note that `localhost` is now also allowed by the spec (as long as the UA
guarantees that it resolves to a loopback address), but Chrome does not
support it so I chose not to test it either.

We should ideally also test the IPv6 loopback address (`::1`), but our testing
infrastructure does not seem to like IPv6 addresses. I'll investigate and
submit a follow-up PR for that.

### CODE REVIEW

**Comment by poiru** [comment]
cc @mikewest

**Comment by w3c-bots** [comment]
*This report has been truncated because the total content is 635514 characters in length, which is in excess of GitHub.com's limit for comments (65536 characters).

 [View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/217300794)

# Firefox (nightly channel) #
Testing web-platform-tests at revision bc19d3017752b041b653976cb4e0543c4354daca
Using browser at version BuildID 20170331102157; SourceStamp 8df9fabf2587b7020889755acb9e75b664fe13cf
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>326 tests ran</summary>
<details>
<summary><a href="https://w3c-test.org/submissions/5304/mixed-content/allowed/http-csp/loopback-ipv4-http/audio-tag/top-level/keep-scheme-redirect/loopback-allowed.https.html">/mixed-content/allowed/http-csp/loopback-ipv4-http/audio-tag/top-level/keep-scheme-redirect/loopback-allowed.https.html</a></summary>


|                                                                                                                                                                                            Subtest                                                                                                                                                                                             | Results |                                              Messages                                             |
|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------|---------------------------------------------------------------------------------------------------|
|                                                                                                                                                                                                                                                                                                                                                                                                | OK      |                                                                                                   |
| `opt_in_method: http-csp\n                                 origin: loopback-ipv4-http\n                                 source_scheme: https\n                                 context_nesting: top-level\n                                 redirection: keep-scheme-redirect\n                                 subresource: audio-tag\n                                 expectation: allowed` | FAIL    | `assert_equals: The triggered event should match 'allowed'. expected "allowed" but got "blocked"` |

</details>
<details>
<summary><a href="https://w3c-test.org/submissions/5304/mixed-content/allowed/http-csp/loopback-ipv4-http/audio-tag/top-level/no-redirect/loopback-allowed.https.html">/mixed-content/allowed/http-csp/loopback-ipv4-http/audio-tag/top-level/no-redirect/loopback-allowed.https.html</a></summary>


|                                                                                                                                                                                        Subtest                                                                                                                                                                                        | Results |                                              Messages                                             |
|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------|---------------------------------------------------------------------------------------------------|
|                                                                                                                                                                                                                                                                                                                                                                                       | OK      |                                                                                                   |
| `opt_in_method: http-csp\n                                 origin: loopback-ipv4-http\n                                 source_scheme: https\n                                 context_nesting: top-level\n                                 redirection: no-redirect\n                                 subresource: audio-tag\n                                 expectation: allowed` | FAIL    | `assert_equals: The triggered event should match 'allowed'. expected "allowed" but got "blocked"` |

</details>
<details>
<summary><a href="https://w3c-test.org/submissions/5304/mixed-content/allowed/http-csp/loopback-ipv4-http/fetch-request/top-level/keep-scheme-redirect/loopback-allowed.https.html">/mixed-content/allowed/http-csp/loopback-ipv4-http/fetch-request/top-level/keep-scheme-redirect/loopback-allowed.https.html</a></summary>


|                                                                                                                                                                                              Subtest                                                                                                                                                                                               | Results |                                              Messages                                             |
|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------|---------------------------------------------------------------------------------------------------|
|                                                                                                                                                                                                                                                                                                                                                                                                    | OK      |                                                                                                   |
| `opt_in_method: http-csp\n                                 origin: loopback-ipv4-http\n                                 source_scheme: https\n                                 context_nesting: top-level\n                                 redirection: keep-scheme-redirect\n                                 subresource: fetch-request\n                                 expectation: allowed` | FAIL    | `assert_equals: The triggered event should match 'allowed'. expected "allowed" but got "blocked"` |

</details>
<details>
<summary><a href="https://w3c-test.org/submissions/5304/mixed-content/allowed/http-csp/loopback-ipv4-http/fetch-request/top-level/no-redirect/loopback-allowed.https.html">/mixed-content/allowed/http-csp/loopback-ipv4-http/fetch-request/top-level/no-redirect/loopback-allowed.https.html</a></summary>


|                                                                                                                                                                                          Subtest                                                                                                                                                                                          | Results |                                              Messages                                             |
|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------|---------------------------------------------------------------------------------------------------|
|                                                                                                                                                                                                                                                                                                                                                                                           | OK      |                                                                                                   |
| `opt_in_method: http-csp\n                                 origin: loopback-ipv4-http\n                                 source_scheme: https\n                                 context_nesting: top-level\n                                 redirection: no-redirect\n                                 subresource: fetch-request\n                                 expectation: allowed` | FAIL    | `assert_equals: The triggered event should match 'allowed'. expected "allowed" but got "blocked"` |

</details>
<details>
<summary><a href="https://w3c-test.org/submissions/5304/mixed-content/allowed/http-csp/loopback-ipv4-http/form-tag/top-level/keep-scheme-redirect/loopback-allowed.https.html">/mixed-content/allowed/http-csp/loopback-ipv4-http/form-tag/top-level/keep-scheme-redirect/loopback-allowed.https.html</a></summary>


| Subtest | Results | Messages |
|---------|---------|----------|
|         | TIMEOUT |          |

</details>
<details>
<summary><a href="https://w3c-test.org/submissions/5304/mixed-content/allowed/http-csp/loopback-ipv4-http/form-tag/top-level/no-redirect/loopback-allowed.https.html">/mixed-content/allowed/http-csp/loopback-ipv4-http/form-tag/top-level/no-redirect/loopback-allowed.https.html</a></summary>


| Subtest | Results | Messages |
|---------|---------|----------|
|         | TIMEOUT |          |

</details>
<details>
<summary><a href="https://w3c-test.org/submissions/5304/mixed-content/allowed/http-csp/loopback-ipv4-http/iframe-tag/top-level/keep-scheme-redirect/loopback-allowed.https.html">/mixed-content/allowed/http-csp/loopback-ipv4-http/iframe-tag/top-level/keep-scheme-redirect/loopback-allowed.https.html</a></summary>


|                                                                                                                                                                                             Subtest                                                                                                                                                                                             | Results |                                              Messages                                             |
|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------|---------------------------------------------------------------------------------------------------|
|                                                                                                                                                                                                                                                                                                                                                                                                 | OK      |                                                                                                   |
| `opt_in_method: http-csp\n                                 origin: loopback-ipv4-http\n                                 source_scheme: https\n                                 context_nesting: top-level\n                                 redirection: keep-scheme-redirect\n                                 subresource: iframe-tag\n                                 expectation: allowed` | FAIL    | `assert_equals: The triggered event should match 'allowed'. expected "allowed" but got "blocked"` |

</details>
<details>
<summary><a href="https://w3c-test.org/submissions/5304/mixed-content/allowed/http-csp/loopback-ipv4-http/iframe-tag/top-level/no-redirect/loopback-allowed.https.html">/mixed-content/allowed/http-csp/loopback-ipv4-http/iframe-tag/top-level/no-redirect/loopback-allowed.https.html</a></summary>


|                                                                                                                                                                                        Subtest                                                                                                                                                                                         | Results |                                              Messages                                             |
|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------|---------------------------------------------------------------------------------------------------|
|                                                                                                                                                                                                                                                                                                                                                                                        | OK      |                                                                                                   |
| `opt_in_method: http-csp\n                                 origin: loopback-ipv4-http\n                                 source_scheme: https\n                                 context_nesting: top-level\n                                 redirection: no-redirect\n                                 subresource: iframe-tag\n                                 expectation: allowed` | FAIL    | `assert_equals: The triggered event should match 'allowed'. expected "allowed" but got "blocked"` |

</details>
<details>
<summary><a href="https://w3c-test.org/submissions/5304/mixed-content/allowed/http-csp/loopback-ipv4-http/img-tag/top-level/keep-scheme-redirect/loopback-allowed.https.html">/mixed-content/allowed/http-csp/loopback-ipv4-http/img-tag/top-level/keep-scheme-redirect/loopback-allowed.https.html</a></summary>


|                                                                                                                                                                                           Subtest                                                                                                                                                                                            | Results |                                              Messages                                             |
|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------|---------------------------------------------------------------------------------------------------|
|                                                                                                                                                                                                                                                                                                                                                                                              | OK      |                                                                                                   |
| `opt_in_method: http-csp\n                                 origin: loopback-ipv4-http\n                                 source_scheme: https\n                                 context_nesting: top-level\n                                 redirection: keep-scheme-redirect\n                                 subresource: img-tag\n                                 expectation: allowed` | FAIL    | `assert_equals: The triggered event should match 'allowed'. expected "allowed" but got "blocked"` |

</details>
<details>
<summary><a href="https://w3c-test.org/submissions/5304/mixed-content/allowed/http-csp/loopback-ipv4-http/img-tag/top-level/no-redirect/loopback-allowed.https.html">/mixed-content/allowed/http-csp/loopback-ipv4-http/img-tag/top-level/no-redirect/loopback-allowed.https.html</a></summary>


|                                                                                                                                                                                       Subtest                                                                                                                                                                                       | Results |                                              Messages                                             |
|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------|---------------------------------------------------------------------------------------------------|
|                                                                                                                                                                                                                                                                                                                                                                                     | OK      |                                                                                                   |
| `opt_in_method: http-csp\n                                 origin: loopback-ipv4-http\n                                 source_scheme: https\n                                 context_nesting: top-level\n                                 redirection: no-redirect\n                                 subresource: img-tag\n                                 expectation: allowed` | FAIL    | `assert_equals: The triggered event should match 'allowed'. expected "allowed" but got "blocked"` |

</details>
<details>
<summary><a href="https://w3c-test.org/submissions/5304/mixed-content/allowed/http-csp/loopback-ipv4-http/link-css-tag/top-level/keep-scheme-redirect/loopback-allowed.https.html">/mixed-content/allowed/http-csp/loopback-ipv4-http/link-css-tag/top-level/keep-scheme-redirect/loopback-allowed.https.html</a></summary>


|                                                                                                                                                                                              Subtest                                                                                                                                                                                              | Results |                                              Messages                                             |
|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------|---------------------------------------------------------------------------------------------------|
|                                                                                                                                                                                                                                                                                                                                                                                                   | OK      |                                                                                                   |
| `opt_in_method: http-csp\n                                 origin: loopback-ipv4-http\n                                 source_scheme: https\n                                 context_nesting: top-level\n                                 redirection: keep-scheme-redirect\n                                 subresource: link-css-tag\n                                 expectation: allowed` | FAIL    | `assert_equals: The triggered event should match 'allowed'. expected "allowed" but got "blocked"` |

</details>
<details>
<summary><a href="https://w3c-test.org/submissions/5304/mixed-content/allowed/http-csp/loopback-ipv4-http/link-css-tag/top-level/no-redirect/loopback-allowed.https.html">/mixed-content/allowed/http-csp/loopback-ipv4-http/link-css-tag/top-level/no-redirect/loopback-allowed.https.html</a></summary>


|                                                                                                                                                                                         Subtest                                                                                                                                                                                          | Results |                                              Messages                                             |
|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------|---------------------------------------------------------------------------------------------------|
|                                                                                                                                                                                                                                                                                                                                                                                          | OK      |                                                                                                   |
| `opt_in_method: http-csp\n                                 origin: loopback-ipv4-http\n                                 source_scheme: https\n                                 context_nesting: top-level\n                                 redirection: no-redirect\n                                 subresource: link-css-tag\n                                 expectation: allowed` | FAIL    | `assert_equals: The triggered event should match 'allowed'. expected "allowed" but got "blocked"` |

</details>
<details>
<summary><a href="https://w3c-test.org/submissions/5304/mixed-content/allowed/http-csp/loopback-ipv4-http/link-prefetch-tag/top-level/keep-scheme-redirect/loopback-allowed.https.html">/mixed-content/allowed/http-csp/loopback-ipv4-http/link-prefetch-tag/top-level/keep-scheme-redirect/loopback-allowed.https.html</a></summary>


|                                                                                                                                                                                                Subtest                                                                                                                                                                                                 | Results |                                              Messages                                             |
|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------|---------------------------------------------------------------------------------------------------|
|                                                                                                                                                                                                                                                                                                                                                                                                        | OK      |                                                                                                   |
| `opt_in_method: http-csp\n                                 origin: loopback-ipv4-http\n                                 source_scheme: https\n                                 context_nesting: top-level\n                                 redirection: keep-scheme-redirect\n                                 subresource: link-prefetch-tag\n                                 expectation: allowed` | FAIL    | `assert_equals: The triggered event should match 'allowed'. expected "allowed" but got "blocked"` |

</details>
<details>
<summary><a href="https://w3c-test.org/submissions/5304/mixed-content/allowed/http-csp/loopback-ipv4-http/link-prefetch-tag/top-level/no-redirect/loopback-allowed.https.html">/mixed-content/allowed/http-csp/loopback-ipv4-http/link-prefetch-tag/top-level/no-redirect/loopback-allowed.https.html</a></summary>


|                                                                                                                                                                                            Subtest                                                                                                                                                                                            | Results |                                              Messages                                             |
|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------|---------------------------------------------------------------------------------------------------|
|                                                                                                                                                                                                                                                                                                                                                                                               | OK      |                                                                                                   |
| `opt_in_method: http-csp\n                                 origin: loopback-ipv4-http\n                                 source_scheme: https\n                                 context_nesting: top-level\n                                 redirection: no-redirect\n                                 subresource: link-prefetch-tag\n                                 expectation: allowed` | FAIL    | `assert_equals: The triggered event should match 'allowed'. expected "allowed" but got "blocked"` |

</details>
<details>
<summary><a href="https://w3c-test.org/submissions/5304/mixed-content/allowed/http-csp/loopback-ipv4-http/object-tag/top-level/keep-scheme-redirect/loopback-allowed.https.html">/mixed-content/allowed/http-csp/loopback-ipv4-http/object-tag/top-level/keep-scheme-redirect/loopback-allowed.https.html</a></summary>


|                                                                                                                                                                                             Subtest                                                                                                                                                                                             | Results |                                              Messages                                             |
|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------|---------------------------------------------------------------------------------------------------|
|                                                                                                                                                                                                                                                                                                                                                                                                 | OK      |                                                                                                   |
| `opt_in_method: http-csp\n                                 origin: loopback-ipv4-http\n                                 source_scheme: https\n                                 context_nesting: top-level\n                                 redirection: keep-scheme-redirect\n                                 subresource: object-tag\n                                 expectation: allowed` | FAIL    | `assert_equals: The triggered event should match 'allowed'. expected "allowed" but got "blocked"` |

</details>
<details>
<summary><a href="https://w3c-test.org/submissions/5304/mixed-content/allowed/http-csp/loopback-ipv4-http/object-tag/top-level/no-redirect/loopback-allowed.https.html">/mixed-content/allowed/http-csp/loopback-ipv4-http/object-tag/top-level/no-redirect/loopback-allowed.https.html</a></summary>


|                                                                                                                                                                                        Subtest                                                                                                                                                                                         | Results |                                              Messages                                             |
|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------|---------------------------------------------------------------------------------------------------|
|                                                                                                                                                                                                                                                                                                                                                                                        | OK      |                                                                                                   |
| `opt_in_method: http-csp\n                                 origin: loopback-ipv4-http\n                                 source_scheme: https\n                                 context_nesting: top-level\n                                 redirection: no-redirect\n                                 subresource: object-tag\n                                 expectation: allowed` | FAIL    | `assert_equals: The triggered event should match 'allowed'. expected "allowed" but got "blocked"` |

</details>
<details>
<summary><a href="https://w3c-test.org/submissions/5304/mixed-content/allowed/http-csp/loopback-ipv4-http/picture-tag/top-level/keep-scheme-redirect/loopback-allowed.https.html">/mixed-content/allowed/http-csp/loopback-ipv4-http/picture-tag/top-level/keep-scheme-redirect/loopback-allowed.https.html</a></summary>


|                                                                                                                                                                                             Subtest                                                                                                                                                                                              | Results |                                              Messages                                             |
|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------|---------------------------------------------------------------------------------------------------|
|                                                                                                                                                                                                                                                                                                                                                                                                  | OK      |                                                                                                   |
| `opt_in_method: http-csp\n                                 origin: loopback-ipv4-http\n                                 source_scheme: https\n                                 context_nesting: top-level\n                                 redirection: keep-scheme-redirect\n                                 subresource: picture-tag\n                                 expectation: allowed` | FAIL    | `assert_equals: The triggered event should match 'allowed'. expected "allowed" but got "blocked"` |

</details>
<details>
<summary><a href="https://w3c-test.org/submissions/5304/mixed-content/allowed/http-csp/loopback-ipv4-http/picture-tag/top-level/no-redirect/loopback-allowed.https.html">/mixed-content/allowed/http-csp/loopback-ipv4-http/picture-tag/top-level/no-redirect/loopback-allowed.https.html</a></summary>


|                                                                                                                                                                                         Subtest                                                                                                                                                                                         | Results |                                              Messages                                             |
|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------|---------------------------------------------------------------------------------------------------|
|                                                                                                                                                                                                                                                                                                                                                                                         | OK      |                                                                                                   |
| `opt_in_method: http-csp\n                                 origin: loopback-ipv4-http\n                                 source_scheme: https\n                                 context_nesting: top-level\n                                 redirection: no-redirect\n                                 subresource: picture-tag\n                                 expectation: allowed` | FAIL    | `assert_equals: The triggered event should match 'allowed'. expected "allowed" but got "blocked"` |

</details>
<details>
<summary><a href="https://w3c-test.org/submissions/5304/mixed-content/allowed/http-csp/loopback-ipv4-http/script-tag/top-level/keep-scheme-redirect/loopback-allowed.https.html">/mixed-content/allowed/http-csp/loopback-ipv4-http/script-tag/top-level/keep-scheme-redirect/loopback-allowed.https.html</a></summary>


|                                                                                                                                                                                             Subtest                                                                                                                                                                                             | Results |                                              Messages                                             |
|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------|---------------------------------------------------------------------------------------------------|
|                                                                                                                                                                                                                                                                                                                                                                                                 | OK      |                                                                                                   |
| `opt_in_method: http-csp\n                                 origin: loopback-ipv4-http\n                                 source_scheme: https\n                                 context_nesting: top-level\n                                 redirection: keep-scheme-redirect\n                                 subresource: script-tag\n                                 expectation: allowed` | FAIL    | `assert_equals: The triggered event should match 'allowed'. expected "allowed" but got "blocked"` |

</details>
<details>
<summary><a href="https://w3c-test.org/submissions/5304/mixed-content/allowed/http-csp/loopback-ipv4-http/script-tag/top-level/no-redirect/loopback-allowed.https.html">/mixed-content/allowed/http-csp/loopback-ipv4-http/script-tag/top-level/no-redirect/loopback-allowed.https.html</a></summary>


|                                                                                                                                                                                        Subtest                                                                                                                                                                                         | Results |                                              Messages                                             |
|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------|---------------------------------------------------------------------------------------------------|
|                                                                                                                                                                                                                                                                                                                                                                                        | OK      |                                                                                                   |
| `opt_in_method: http-csp\n                                 origin: loopback-ipv4-http\n                                 source_scheme: https\n                                 context_nesting: top-level\n                                 redirection: no-redirect\n                                 subresource: script-tag\n                                 expectation: allowed` | FAIL    | `assert_equals: The triggered event should match 'allowed'. expected "allowed" but got "blocked"` |

</details>
<details>
<summary><a href="https://w3c-test.org/submissions/5304/mixed-content/allowed/http-csp/loopback-ipv4-http/video-tag/top-level/keep-scheme-redirect/loopback-allowed.https.html">/mixed-content/allowed/http-csp/loopback-ipv4-http/video-tag/top-level/keep-scheme-redirect/loopback-allowed.https.html</a></summary>


|                                                                                                                                                                                            Subtest                                                                                                                                                                                             | Results |                                              Messages                                             |
|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------|---------------------------------------------------------------------------------------------------|
|                                                                                                                                                                                                                                                                                                                                                                                                | OK      |                                                                                                   |
| `opt_in_method: http-csp\n                                 origin: loopback-ipv4-http\n                                 source_scheme: https\n                                 context_nesting: top-level\n                                 redirection: keep-scheme-redirect\n                                 subresource: video-tag\n                                 expectation: allowed` | FAIL    | `assert_equals: The triggered event should match 'allowed'. expected "allowed" but got "blocked"` |

</details>
<details>
<summary><a href="https://w3c-test.org/submissions/5304/mixed-content/allowed/http-csp/loopback-ipv4-http/video-tag/top-level/no-redirect/loopback-allowed.https.html">/mixed-content/allowed/http-csp/loopback-ipv4-http/video-tag/top-level/no-redirect/loopback-allowed.https.html</a></summary>


|                                                                                                                                                                                        Subtest                                                                                                                                                                                        | Results |                                              Messages                                             |
|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------|---------------------------------------------------------------------------------------------------|
|                                                                                                                                                                                                                                                                                                                                                                                       | OK      |                                                                                                   |
| `opt_in_method: http-csp\n                                 origin: loopback-ipv4-http\n                                 source_scheme: https\n                                 context_nesting: top-level\n                                 redirection: no-redirect\n                                 subresource: video-tag\n                                 expectation: allowed` | FAIL    | `assert_equals: The triggered event should match 'allowed'. expected "allowed" but got "blocked"` |

</details>
<details>
<summary><a href="https://w3c-test.org/submissions/5304/mixed-content/allowed/http-csp/loopback-ipv4-http/xhr-request/top-level/keep-scheme-redirect/loopback-allowed.https.html">/mixed-content/allowed/http-csp/loopback-ipv4-http/xhr-request/top-level/keep-scheme-redirect/loopback-allowed.https.html</a></summary>


|                                                                                                                                                                                             Subtest                                                                                                                                                                                              | Results |                                              Messages                                             |
|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------|---------------------------------------------------------------------------------------------------|
|                                                                                                                                                                                                                                                                                                                                                                                                  | OK      |                                                                                                   |
| `opt_in_method: http-csp\n                                 origin: loopback-ipv4-http\n                                 source_scheme: https\n                                 context_nesting: top-level\n                                 redirection: keep-scheme-redirect\n                                 subresource: xhr-request\n                                 expectation: allowed` | FAIL    | `assert_equals: The triggered event should match 'allowed'. expected "allowed" but got "blocked"` |

</details>
<details>
<summary><a href="https://w3c-test.org/submissions/5304/mixed-content/allowed/http-csp/loopback-ipv4-http/xhr-request/top-level/no-redirect/loopback-allowed.https.html">/mixed-content/allowed/http-csp/loopback-ipv4-http/xhr-request/top-level/no-redirect/loopback-allowed.https.html</a></summary>


|                                                                                                                                                                                         Subtest                                                                                                                                                                                         | Results |                                              Messages                                             |
|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------|---------------------------------------------------------------------------------------------------|
|                                                                                                                                                                                                                                                                                                                                                                                         | OK      |                                                                                                   |
| `opt_in_method: http-csp\n                                 origin: loopback-ipv4-http\n                                 source_scheme: https\n                                 context_nesting: top-level\n                                 redirection: no-redirect\n                                 subresource: xhr-request\n                                 expectation: allowed` | FAIL    | `assert_equals: The triggered event should match 'allowed'. expected "allowed" but got "blocked"` |

</details>
<details>
<summary><a href="https://w3c-test.org/submissions/5304/mixed-content/allowed/http-csp/same-host-https/audio-tag/top-level/keep-scheme-redirect/allowed.https.html">/mixed-content/allowed/http-csp/same-host-https/audio-tag/top-level/keep-scheme-redirect/allowed.https.html</a></summary>


|                                                                                                                                                                                           Subtest                                                                                                                                                                                           | Results |                                              Messages                                             |
|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------|---------------------------------------------------------------------------------------------------|
|                                                                                                                                                                                                                                                                                                                                                                                             | OK      |                                                                                                   |
| `opt_in_method: http-csp\n                                 origin: same-host-https\n                                 source_scheme: https\n                                 context_nesting: top-level\n                                 redirection: keep-scheme-redirect\n                                 subresource: audio-tag\n                                 expectation: allowed` | FAIL    | `assert_equals: The triggered event should match 'allowed'. expected "allowed" but got "blocked"` |

</details>
<details>
<summary><a href="https://w3c-test.org/submissions/5304/mixed-content/allowed/http-csp/same-host-https/audio-tag/top-level/no-redirect/allowed.https.html">/mixed-content/allowed/http-csp/same-host-https/audio-tag/top-level/no-redirect/allowed.https.html</a></summary>


|                                                                                                                                                                                      Subtest                                                                                                                                                                                       | Results |                                              Messages                                             |
|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------|---------------------------------------------------------------------------------------------------|
|                                                                                                                                                                                                                                                                                                                                                                                    | OK      |                                                                                                   |
| `opt_in_method: http-csp\n                                 origin: same-host-https\n                                 source_scheme: https\n                                 context_nesting: top-level\n                                 redirection: no-redirect\n                                 subresource: audio-tag\n                                 expectation: allowed` | FAIL    | `assert_equals: The triggered event should match 'allowed'. expected "allowed" but got "blocked"` |

</details>
<details>
<summary><a href="https://w3c-test.org/submissions/5304/mixed-content/allowed/http-csp/same-host-https/fetch-request/top-level/keep-scheme-redirect/allowed.https.html">/mixed-content/allowed/http-csp/same-host-https/fetch-request/top-level/keep-scheme-redirect/allowed.https.html</a></summary>


|                                                                                                                                                                                             Subtest                                                                                                                                                                                             | Results | Messages |
|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------|----------|
|                                                                                                                                                                                                                                                                                                                                                                                                 | OK      |          |
| `opt_in_method: http-csp\n                                 origin: same-host-https\n                                 source_scheme: https\n                                 context_nesting: top-level\n                                 redirection: keep-scheme-redirect\n                                 subresource: fetch-request\n                                 expectation: allowed` | PASS    |          |

</details>
<details>
<summary><a href="https://w3c-test.org/submissions/5304/mixed-content/allowed/http-csp/same-host-https/fetch-request/top-level/no-redirect/allowed.https.html">/mixed-content/allowed/http-csp/same-host-https/fetch-request/top-level/no-redirect/allowed.https.html</a></summary>


|                                                                                                                                                                                        Subtest                                                                                                                                                                                         | Results | Messages |
|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------|----------|
|                                                                                                                                                                                                                                                                                                                                                                                        | OK      |          |
| `opt_in_method: http-csp\n                                 origin: same-host-https\n                                 source_scheme: https\n                                 context_nesting: top-level\n                                 redirection: no-redirect\n                                 subresource: fetch-request\n                                 expectation: allowed` | PASS    |          |

</details>
<details>
<summary><a href="https://w3c-test.org/submissions/5304/mixed-content/allowed/http-csp/same-host-https/form-tag/top-level/keep-scheme-redirect/allowed.https.html">/mixed-content/allowed/http-csp/same-host-https/form-tag/top-level/keep-scheme-redirect/allowed.https.html</a></summary>


|                                                                                                                                                                                          Subtest                                                                                                                                                                                           | Results | Messages |
|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------|----------|
|                                                                                                                                                                                                                                                                                                                                                                                            | OK      |          |
| `opt_in_method: http-csp\n                                 origin: same-host-https\n                                 source_scheme: https\n                                 context_nesting: top-level\n                                 redirection: keep-scheme-redirect\n                                 subresource: form-tag\n                                 expectation: allowed` | PASS    |          |

</details>
<details>
<summary><a href="https://w3c-test.org/submissions/5304/mixed-content/allowed/http-csp/same-host-https/form-tag/top-level/no-redirect/allowed.https.html">/mixed-content/allowed/http-csp/same-host-https/form-tag/top-level/no-redirect/allowed.https.html</a></summary>


|                                                                                                                                                                                      Subtest                                                                                                                                                                                      | Results | Messages |
|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------|----------|
|                                                                                                                                                                                                                                                                                                                                                                                   | OK      |          |
| `opt_in_method: http-csp\n                                 origin: same-host-https\n                                 source_scheme: https\n                                 context_nesting: top-level\n                                 redirection: no-redirect\n                                 subresource: form-tag\n                                 expectation: allowed` | PASS    |          |

</details>
<details>
<summary><a href="https://w3c-test.org/submissions/5304/mixed-content/allowed/http-csp/same-host-https/iframe-tag/top-level/keep-scheme-redirect/allowed.https.html">/mixed-content/allowed/http-csp/same-host-https/iframe-tag/top-level/keep-scheme-redirect/allowed.https.html</a></summary>


|                                                                                                                                                                                           Subtest                                                                                                                                                                                            | Results | Messages |
|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------|----------|
|

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/217300795)

# Chrome (unstable channel) #
Testing web-platform-tests at revision bc19d3017752b041b653976cb4e0543c4354daca
Using browser at version 59.0.3053.3 dev
Starting 10 test iterations

**Comment by poiru** [comment]
@Ms2ger Could you help me find someone to review this?

**Comment by Ms2ger** [comment]
@kristijanburnik and @mikewest seem good targets, and if not they should probably be able to find someone.

File: mixed-content/generic/mixed-content-test-case.js [hunk 1]
```diff
@@ -23,6 +23,7 @@ function MixedContentTestCase(scenario, description, sanityChecker) {
 
   var sameOriginHost = location.hostname;
   var crossOriginHost = "{{domains[www1]}}";
+  var localhostIPv4Host = "127.0.0.1";
```
Comment (mikewest) [inline mixed-content/generic/mixed-content-test-case.js]: I don't think this is going to work in general. @jgraham will have thoughts, I'm sure, but I don't think we can make the assumption that 127.0.0.1 is actually running a server. It won't be running a server on `w3c-test.org`, for instance.

Comment (poiru) [inline mixed-content/generic/mixed-content-test-case.js]: @jgraham ping ^

**Pull request closed without merging**